File: bin/benchmark/string.py
import time

t1 = time.time()

def loop(n):
    ret = ""
    last = ""

    for i in range(2, n):
        ret = last + "ret" + str(i)
        last = "last" + str(i)
    return ret


N = 1000000

print(loop(N),time.time() - t1)

File: bin/benchmark/string.lua
local x = os.clock()
function loop(n)
    local ret = ""
    local last = ""

    for i = 2, n do
        ret = last.."ret"..i
        last = "last"..i
    end
    return ret
end

local N = 1000000

print(loop(N))
print(string.format("elapsed time: %.2f\n", os.clock() - x))